iavl/internal/unsafe_bytes.go

```diff
@@ -0,0 +1,56 @@
+package internal
+
+// UnsafeBytes wraps a byte slice that may or not be a direct reference to
+// a memory-mapped file.
+// Generally, an unsafe byte slice cannot be expected to live longer than the
+// Pin on the object it was obtained from.
+// As long as it is pinned, it is safe to use the UnsafeBytes() method to get
+// the underlying byte slice without copying.
+// If the byte slice needs to be retained beyond the Pin's lifetime, the
+// SafeCopy() method must be used to get a safe copy of the byte slice.
+type UnsafeBytes struct {
+	bz   []byte
+	safe bool
+}
+
+// WrapUnsafeBytes wraps an unsafe byte slice as UnsafeBytes, indicating that
+// it is unsafe to use without copying.
+// Use this method when you are wrapping a byte slice obtained from a memory-mapped file.
+func WrapUnsafeBytes(bz []byte) UnsafeBytes {
+	return UnsafeBytes{bz: bz, safe: false}
+}
+
+// WrapSafeBytes wraps a safe byte slice as UnsafeBytes, indicating that
+// it is safe to use without copying.
+// Use this method when you are wrapping a byte slice that is known to be safe,
+// e.g., a byte slice allocated in regular garbage-collected memory.
+func WrapSafeBytes(bz []byte) UnsafeBytes {
+	return UnsafeBytes{bz: bz, safe: true}
```

Comment:
seems confusing that we have a type that represents both safe and unsafe bytes but is housed under unsafe.

would it make sense to have two types SafeBytes and UnsafeBytes? i guess that may not work cuz the point seems that this type is more like a Maybe Maybe Not Safe Bytes